### PR TITLE
Add tests for Workers. Tests #591

### PR DESCRIPTION
**Pull request by alippai**

### CODE REVIEW

File: script/test.js [hunk 1]
```diff
@@ -51,6 +51,10 @@ if (process.env.HARMONY) {
   nodeArgs.push("--harmony")
 }
 
+if (process.version.startsWith("v10.") || process.version.startsWith("v11.")) {
+  nodeArgs.push("--experimental-worker")
+}
+
```
Comment (jdalton) [inline script/test.js]: ☝️ You can use `semver` for this check (it's a dev dep we use for just this kind of sniffing for our tests).
```js
const SemVer = require("semver")
// then later
SemVer.satisfies(process.version, ">10.5.0")
```

File: test/misc-tests.js [hunk 2]
```diff
@@ -1566,4 +1566,44 @@ describe("miscellaneous tests", () => {
       import("./fixture/source/source-mapping-url-string.mjs")
     )
   })
+
+    // Test for https://github.com/standard-things/esm/issues/591
+  describe("Workers support", () => {
+    before(function ()  {
+      if (!process.version.startsWith("v10.") && !process.version.startsWith("v11.")) {
+        this.skip()
+      }
```
Comment (jdalton) [inline test/misc-tests.js]: At the top of the misc-test.js file you'll see other `canDoThisOrThat` vars. You can add one for
```
const canTestWorkerThreads = SemVer.satisfies(process.version, ">10.5.0")
```

**Comment by alippai** [comment]
@jdalton the new commit addresses your review comments. Thanks a lot!

**Comment by jdalton** [comment]
Merged as https://github.com/standard-things/esm/commit/49918c5a2e53b4d58cdae8665d7741fd33e4e133.

**Pull request closed without merging**